server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2850,7 +2850,7 @@ public synchronized void afterMerge(OnGoingMerge merge) {
                     @Override
                     public void onFailure(Exception e) {
                         if (isClosed.get() == false) {
-                            logger.warn("failed to flush after merge has finished");
+                            logger.warn("failed to flush after merge has finished", e);
                         }
```

Comment:
I wonder if we should silently swallow the error if we are shutting down? 
May be at least log it at debug level?

Reply:
debug on shutdown would be fine with me too